src/EventSubscriber/Api/Oauth2ResponseSubscriber.php

```diff
@@ -100,4 +93,14 @@ public function extractUserIdFromJwt(string $jwt): string
             ->claims()
             ->get('sub');
     }
+
+    private function findUser(TokenRequestResolveEvent $event): ?User
+    {
+        $username = $this->extractUserIdFromResponse($event->getResponse());
+        if (!$username) {
+            return null;
+        }
+
+        return $this->repository->findByUsername($username);
```

Comment:
`return username ? $this->repository->findByUsername($username) : null;`

Reply:
```php
    public function findByUsername($username): ?User
    {
        $query = $this->createQueryBuilder('user')
            ->where('user.username = :username')
            ->setParameter('username', $username)
            ->getQuery();

        try {
            $user = $query->getOneOrNullResult();
        } catch (NonUniqueResultException) {
            return null;
        }

        return $user;
    }
```

Reply:
Et du coup ?